spec/integration/util/windows/user_spec.rb

```diff
@@ -215,5 +215,12 @@ def expect_logon_failure_error(&block)
         expect(Puppet::Util::Windows::User.localsystem?('OtherUser')).to eq(false)
       end
     end
+
+    describe "get_rights" do
+      it "should be empty when given user does not exist" do
+        allow(Puppet::Util::Windows::SID).to receive(:name_to_principal).and_return(nil)
```

Comment:
Should we be more explicit here and also add the expected argument of `name_to_principal`?